test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/DacFx/DacFxserviceTests.cs

```diff
@@ -436,6 +436,163 @@ public async void GenerateDeployPlan()
+                targetDb = SqlTestDb.CreateFromExisting(deployParams.DatabaseName);
+
+                string deployedProc;
+
+                using (SqlConnection conn = new SqlConnection(targetDb.ConnectionString))
+                {
+                    try
+                    {
+                        await conn.OpenAsync();
+                        deployedProc = (string)ReliableConnectionHelper.ExecuteScalar(conn, "SELECT OBJECT_DEFINITION (OBJECT_ID(N'Procedure1'));");
+                    }
+                    finally
+                    {
+                        conn.Close();
+                    }
+                }
+
+                Assert.Contains(deployParams.SqlCommandVariableValues[databaseRefVarName], deployedProc);
+                Assert.Contains(deployParams.SqlCommandVariableValues[filterValueVarName], deployedProc);
+
+                VerifyAndCleanup(extractParams.PackageFilePath);
+            }
+            finally
+            {
+                sourceDb.Cleanup();
+                if (targetDb != null)
+                {
+                    targetDb.Cleanup();
+                }
+            }
+        }
+
+        // <summary>
+        /// Verify that SqlCmdVars are set correctly for a deploy request
+        /// </summary>
+        [Fact]
+        public async void GenerateDeployScriptWithSqlCmdVariables()
+        {
+            const string databaseRefVarName = "DatabaseRef";
+            const string filterValueVarName = "FilterValue";
+
+            string storedProcScript = $@"
+CREATE PROCEDURE [dbo].[Procedure1]
+	@param1 int = 0,
+	@param2 int
+AS
+	SELECT * FROM [$({databaseRefVarName})].[dbo].[Table1] WHERE Type = '$({filterValueVarName})'
+RETURN 0
+";
+
+            // first extract a db to have a dacpac to import later
+            var result = GetLiveAutoCompleteTestObjects();
+            SqlTestDb sourceDb = await SqlTestDb.CreateNewAsync(TestServerType.OnPrem, query: storedProcScript, dbNamePrefix: "DacFxGenerateScriptSqlCmdVarsTest");
+            SqlTestDb targetDb = null;
+            string folderPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "DacFxTest");
+            Directory.CreateDirectory(folderPath);
+
+            try
+            {
+                var extractParams = new ExtractParams
+                {
+                    DatabaseName = sourceDb.DatabaseName,
+                    PackageFilePath = Path.Combine(folderPath, string.Format("{0}.dacpac", sourceDb.DatabaseName)),
+                    ApplicationName = "test",
+                    ApplicationVersion = "1.0.0.0"
+                };
+
+                DacFxService service = new DacFxService();
+                ExtractOperation extractOperation = new ExtractOperation(extractParams, result.ConnectionInfo);
+                service.PerformOperation(extractOperation, TaskExecutionMode.Execute);
```

Comment:
We should to be able take out till here to a common function right? so don't have to write twice?